### PR TITLE
RA-1611 : Follow up commit for diagnosisSets and diagnosisClasses par…

### PR DESCRIPTION
**Pull request by Ruhanga**
…ameters alternate between 'null' and the 'default' from the their GPs

PR : https://issues.openmrs.org/browse/RA-1611

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java [hunk 1]
```diff
@@ -89,7 +89,7 @@
             }
         }
         else {
-            diagnosisSets = emrApiProperties.getDiagnosisSets();
+            diagnosisSets = StringUtils.isNotEmpty(diagnosisConceptSources) || StringUtils.isNotEmpty(diagnosisConceptClasses) ? null : emrApiProperties.getDiagnosisSets();
```
Comment (mks-d) [inline omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java]: We said:
>as soon as we start specifying any attribute, all defaults drop

For the sake of this controller, should the check not be
```java
diagnosisConceptSources != null || diagnosisConceptClasses != null
```
?

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java [hunk 2]
```diff
@@ -105,7 +105,7 @@
             }
         }
         else {
-            sources = emrApiProperties.getConceptSourcesForDiagnosisSearch();
+            sources = StringUtils.isNotEmpty(diagnosisSetUuids) || StringUtils.isNotEmpty(diagnosisConceptClasses) ? null : emrApiProperties.getConceptSourcesForDiagnosisSearch();
```
Comment (mks-d) [inline omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java]: For the sake of this controller, should the check not be
```java
diagnosisSetUuids != null || diagnosisConceptClasses != null
```
?

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java (hunk 1 above)
Comment (Ruhanga) [reply omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java]: Ohh, sorry I didn't clarify that actually spring places a null string/character in the requested parameter's variable. I attempted this earlier and that check failed then did a failover to apache `StringUtils`.

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java (hunk 2 above)
Comment (Ruhanga) [reply omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java]: The same story as above happens here @mks-d.

**Pull request merged**